### ISSUE TITLE
Enhancement: Add option to make tapping a search engine perform the search immediately

### ISSUE THREAD

**Issue by e-t-l** [mozilla-mobile/fenix#22428]
[comment]: # (Please do your best to search for duplicate issues before filing a new issue so we can keep our issue board clean)
[comment]: # (Every issue should have exactly one feature request described in it. Please do not file feedback list tickets as it is difficult to parse them and address their individual points)
[comment]: # (Feature Requests are better when they’re open-ended instead of demanding a specific solution e.g: “I want an easier way to do X” instead of “add Y”)
[comment]: # (Read https://github.com/mozilla-mobile/fenix#i-want-to-file-an-issue for more information)

### What is the user problem or growth opportunity you want to see solved?
It's annoying to type out or paste a search query into the address bar, open the search engine list, select the desired non-default search engine, and then still have to tap Enter to perform the search. 

There should be an option in **Settings > Search** to make the action of selecting a search engine from the list trigger the search immediately (if the address bar is populated) or to navigate to that search engine's home page (if the address bar is empty).
### How do you know that this problem exists today? Why is this important?

As far as "problems" go, perhaps it's not a big one. The added step of tapping Enter to perform a search (or manually navigating to a search engine's homepage) might only add 0.5-1 seconds to the time required to perform a search. But Fenix is a browser, and users make millions of searches. Anything to make UX more intuitive and efficient is a worthwhile benefit (_especially_ because I believe this enhancement would be fairly trivial to implement. I would be more than happy to fork Fenix and try to add it myself, although I could use some assistance in figuring out where to start, i.e. where in this enormous repo would I find the code that handles the search engine selection menu, where would I start for adding a checkbox option to the Search Settings).

### Who will benefit from it?

Any and all Fenix users who use more than one search engine. As the app comes with multiple search engines installed, I would imagine this would functionally be approaching 100% of users.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-18623)

**Comment by mcarare** [comment]
@e-t-l I know it's not exactly what you requested, but when you want to perform a single search using another search engine than the existing one, you can simply type the search engine name, and a suggestion to use that search engine will show up. You can then type your search term and the search will use the selected search engine for performing the search.

**Comment by e-t-l** [comment]
> You can then type your search term and the search will use the selected search engine for performing the search

Unless I'm doing something wrong or misunderstanding you, this is not working for me. If my default search engine is Bing and I type "Google puppies", I get the google suggestion popup after the first word, but after I type the second word and hit enter, it performs a Bing search for the terms "google puppies". The user still has to tap the search engine suggestion popup in order to switch search engines.  So, it's not actually saving any time.

**Comment by mcarare** [comment]
@e-t-l In your case, just type google (it actually shows up after just typing "g"), not the search term and the first suggestion in the list would be "Search Google" with the description "Search directly from the address bar". Tap on that and only after that type your search term.

**Comment by e-t-l** [comment]
Yes, those are the steps I described in my comment. This is the UX this issue is intended to improve.

**Comment by basings** [comment]
firefox desktop has the same behavior as you described fore fenix, you need multiple clicks to search on a different engine

**Comment by e-t-l** [comment]
That's true, the desktop search UX could also stand to benefit from this enhancement. However, it's less of an issue on desktop because desktop users can make use of search keywords.

**Comment by boek** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1813696

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
